pkg/expression/schema.go

```diff
@@ -80,6 +80,24 @@ func (s *Schema) Clone() *Schema {
 	return schema
 }
 
+// ExprReferenceSchema checks if any column of this expression are from the schema.
+func ExprReferenceSchema(expr Expression, schema *Schema) bool {
+	switch v := expr.(type) {
+	case *Column:
+		return schema.Contains(v)
+	case *ScalarFunction:
+		for _, arg := range v.GetArgs() {
+			if ExprReferenceSchema(arg, schema) {
+				return true
+			}
+		}
+		return false
+	case *CorrelatedColumn, *Constant:
```

Comment:
Should we consider `CorrelatedColumn.Column` in this case?

Reply:
> Should we consider `CorrelatedColumn.Column` in this case?

It could be risky to do so since the nullability of correlated  column on the inner side is hard t o figure out.